release/nightly_tests/dataset/read_from_uris_benchmark.py

```diff
@@ -0,0 +1,45 @@
+import io
+
+import boto3
+import numpy as np
+from PIL import Image
+
+import ray
+from ray.data import ActorPoolStrategy
+from benchmark import Benchmark
+
+BUCKET = "anyscale-imagenet"
+# This Parquet file contains the keys of images in the 'anyscale-imagenet' bucket.
+METADATA_PATH = "s3://anyscale-imagenet/metadata.parquet"
+
+
+def main():
+    benchmark = Benchmark("read-from-uris")
+    benchmark.run_fn("imagenet", benchmark_fn)
+    benchmark.write_result()
+
+
+def benchmark_fn():
+    metadata = ray.data.read_parquet(METADATA_PATH)
+    # Assuming there are 80 CPUs and 4 in-flight tasks per actor, we need at least 320
```

Comment:
Should we set this to exactly 320 or should we set this to something in excess of the limit?

Also does the compute config used here have 80 CPUs? Probably unnecessary to setup for the release test but we could programmatically pull these values from somewhere to make this more resilient to those numbers changing and causing a "performance" hit.

Reply:
We could 2x the limit, which is what we currently do in `_autodetect_parallelism`.

> Also does the compute config used here have 80 CPUs?

Not necessarily.

> Probably unnecessary to setup for the release test but we could programmatically pull these values from somewhere to make this more resilient to those numbers changing and causing a "performance" hit.

Yeah, agree. Unfortunately, I don't think there's an API to achieve this for an autoscaling cluster. We've filed a request with the Core team.